packages/babel-plugin-transform-react-jsx/src/index.js

```diff
@@ -2,11 +2,23 @@ import jsx from "@babel/plugin-syntax-jsx";
 import helper from "@babel/helper-builder-react-jsx";
 
 export default function({ types: t }, options) {
-  const pragma = options.pragma || "React.createElement";
-  const throwIfNamespace =
+  const THROW_IF_NAMESPACE =
     options.throwIfNamespace === undefined ? true : !!options.throwIfNamespace;
 
+  const PRAGMA_DEFAULT = options.pragma || "React.createElement";
+  const PRAGMA_FRAG_DEFAULT = options.pragmaFrag || "React.Fragment";
+
   const JSX_ANNOTATION_REGEX = /\*?\s*@jsx\s+([^\s]+)/;
+  const JSX_FRAG_ANNOTATION_REGEX = /\*?\s*@jsxFrag\s+([^\s]+)/;
+
+  // returns a closure that returns an identifier or memberExpression node
+  // based on the given id
+  const createIdentifierParser = (id: string) => () => {
+    return id
+      .split(".")
+      .map(name => t.identifier(name))
+      .reduce((object, property) => t.memberExpression(object, property));
+  };
```

Comment:
This looks suspiciously like `t.buildMatchMemberExpression`

(we really need to document that in handbook 🤔)

Reply:
Ah, nevermind, it's its inverse 😆